code/modules/research/designs/weapon_designs.dm

```diff
@@ -172,159 +122,165 @@
-	name = "50-round drum magazine (.45 ACP)"
-	desc = "A tommygun magazine."
-	id = "tommygunm45"
+/datum/design/seeker
+	name = "Seeker shells"
+	desc = "High caliber armor piercing shells designed for use in the Seeker Rifle."
+	id = "seeker"
 	build_type = FABRICATOR
 	materials = list(/datum/material/iron = 400)
-	build_path = /obj/item/ammo_box/magazine/tommygunm45
+	build_path = /obj/item/ammo_box/magazine/seeker
 	category = list(DCAT_AMMO)
 	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
 
-// Misc guns
+//Rivet
 
-/datum/design/m12g
-	name = "Shotgun magazine (12g buckshot slugs)"
-	desc = "A shotgun magazine."
-	id = "m12g"
+/datum/design/rivet_gun
+	name = "711-MarkCL Rivet Gun"
+	desc = "The latest refinement from Timson Tools' long line of friendly tools."
+	id = "rivet_gun"
 	build_type = FABRICATOR
-	materials = list(/datum/material/iron = 300)
-	build_path = /obj/item/ammo_box/magazine/m12g
-	category = list(DCAT_AMMO)
-	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
+	materials = list(/datum/material/iron = 10000)
+	build_path = /obj/item/gun/ballistic/automatic/pistol/rivet
+	category = list(DCAT_WEAPON)
+	mapload_design_flags = DESIGN_FAB_ENGINEERING | DESIGN_FAB_OMNI //Weapon is weapon, even it's an engineers' weapon
 
-/datum/design/sniper_rounds
-	name = "6-round magazine (.50 BMG)"
-	desc = "A sniper magazine."
-	id = "sniper_rounds"
+/datum/design/rivet
+	name = "Rivet magazine"
+	desc = "A rivet magazine"
+	id = "rivet_gun"
 	build_type = FABRICATOR
 	materials = list(/datum/material/iron = 300)
-	build_path = /obj/item/ammo_box/magazine/sniper_rounds
+	build_path = /obj/item/ammo_box/magazine/rivet
 	category = list(DCAT_AMMO)
-	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
+	mapload_design_flags = DESIGN_FAB_ENGINEERING | DESIGN_FAB_OMNI
 
-/datum/design/recharge
-	name = "Power pack"
-	desc = "A rechargeable, detachable battery that serves as a magazine for laser rifles."
-	id = "recharge"
-	build_type = FABRICATOR
-	materials = list(/datum/material/iron = 1000, /datum/material/silver = 10)
-	build_path = /obj/item/ammo_box/magazine/recharge
-	category = list(DCAT_AMMO)
-	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
+//Underbarrel
 
-/* These ones don't have build_path and materials so you were unable to craft them but recipe was showed
-/datum/design/rubbershot/sec
-	name = "Rubber Slug"
-	id = "sec_rshot"
+/datum/design/shotgun_bola
+	name = "Bola"
+	desc = "A shotgun bola"
+	id = "shotgun_bola"
 	build_type = FABRICATOR
+	materials = list(/datum/material/iron = 300)
+	build_path = /obj/projectile/bullet/shotgun_bola
 	category = list(DCAT_AMMO)
 	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
 
-/datum/design/beanbag_slug/sec
-	name = "Beanbag Slug"
-	id = "sec_beanbag_slug"
+/datum/design/pulse_rocket
+	name = "Rocket"
+	desc = "A pulse rocket"
+	id = "pulse_rocket"
 	build_type = FABRICATOR
+	materials = list(/datum/material/iron = 10000)
```

Comment:
```suggestion
	materials = list(/datum/material/iron = 500, /datum/material/plasma = 1500, /datum/material/uranium=1000)
```
10k iron sounds like a funny meme, but we can make it expensive in different ways.